docs/spec.md

```diff
@@ -206,8 +206,14 @@ Informs the client about available services. Only supported if the server declar
   - `id`: number. The server may reuse ids when services disappear and reappear, but only if the services keeps the exact same name, type, and schema. Clients will use this unique id to cache schema info and deserialization routines.
   - `name`: string
   - `type`: string
-  - `requestSchema`: string
-  - `responseSchema`: string
+  - `requestSchema`: string (schema data only) or object with the fields below. If given as string, schema name and encoding will be derived from `type` (by appending `_Request`) and the previously advertised [supportedEncodings](#server-info).
```

Comment:
I'm confused about what it means to derive the schema encoding from supportedEncodings, because I thought those were message encodings, not schema encodings?

Reply:
I guess we also need a response message encoding in addition to response schema encoding?

Reply:
> I'm confused about what it means to derive the schema encoding from supportedEncodings, because I thought those were message encodings, not schema encodings?

As `serviceCallEncoding` (message encoding) we pick one of the supported encodings from  `supportedEncodings` and then derive the schema encoding from that:

https://github.com/foxglove/studio/blob/2c854ffb9c5a3676fb22818206d581bb1c08cf0f/packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts#L606-L615

It's quite some magic. Maybe we just add `messageEncoding` to the objects that I have added in this PR?

Reply:
Naming is hard... what do you think of:
```
request: {
  encoding: string
  schemaName: string
  schemaEncoding: string
  schema: string
}
```
or 
```
requestEncoding: string
requestSchema: {
  name: string
  encoding: string
  data: string
}
```

or something like that?

cc @defunctzombie for thoughts

Reply:
I am in favor of the 1st option

Reply:
Applied the changes. OK for you like that @jtbandes @jhurliman @defunctzombie ?